spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/operations/AsyncMessagingOperations.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2013-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sqs.operations;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import org.springframework.lang.Nullable;
+import org.springframework.messaging.Message;
+
+/**
+ * Asynchronous messaging operations. Implementations should have defaults for {@link Nullable} fields, and can provide
+ * chained methods interfaces for a more fluent API.
+ *
+ * @author Tomaz Fernandes
+ * @since 3.0
+ */
+public interface AsyncMessagingOperations<T> {
+
+	/**
+	 * Send a {@link Message} to the default endpoint with the provided payload. The payload will be serialized if
+	 * necessary.
+	 *
+	 * @param payload the payload to send.
+	 * @return a {@link CompletableFuture} to be completed with the message's {@link UUID}.
+	 */
+	CompletableFuture<SendResult<T>> sendAsync(T payload);
+
+	/**
+	 * Send a message to the provided endpoint with the provided payload. The payload will be serialized if necessary.
+	 *
+	 * @param endpointName the endpoint to send the message to.
+	 * @param payload the payload to send.
+	 * @return a {@link CompletableFuture} to be completed with the message's {@link UUID}.
+	 */
+	CompletableFuture<SendResult<T>> sendAsync(@Nullable String endpointName, T payload);
```

Comment:
`endpointName` - this essentially means queue name right? Did you give it such a generic name to support queue urls/ARNs or ..?

Reply:
Yeah, well, I still have an idea that this is a meta-framework and hence I'd rather not tie up the generic messaging interfaces specifically to queues.

But I believe the `send(to -> to.queue("myQueue", myPayload))` and `receive(from -> from.queue("myQueue"))` should be the most used methods, and there the implementation is specific to sqs and queues.

What do you think?

Reply:
I think we need to balance it out. If we use `queue` in `to.queue()` it is better to change `endpointName` variable names to `queue`. Right now it is not straightforward what the endpoint name is. Without going into the source code as a user I would assume I must set there queue url.

Reply:
I would stick with `queue` as the word queue does not determine if it is a queue name, ARN, or URL so it leaves us high level of flexibility keeping it easy to understand.